pkg/apis/calico/validation/networkconfig.go

```diff
@@ -0,0 +1,233 @@
+		allErrs = append(allErrs, ValidateIPAutodetectionMethod(*ipv6.AutoDetectionMethod, fldPath.Child("autoDetectionMethod"))...)
+	}
+
+	return allErrs
+}
+
+func ValidateIPAutodetectionMethod(method string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	validOptions := sets.New("first-found", "can-reach", "interface", "skip-interface", "cidr")
+	if method == "" {
+		allErrs = append(allErrs, field.Invalid(fldPath, method, "method cannot be empty"))
+		return allErrs
+	}
+
+	parts := strings.SplitN(method, "=", 2)
+	option := parts[0]
+
+	if !validOptions.Has(option) {
+		allErrs = append(allErrs, field.Invalid(fldPath, method, fmt.Sprintf("invalid option %q, supported options are %v", option, validOptions.UnsortedList())))
+		return allErrs
+	}
+
+	if option == "cidr" {
+		if len(parts) != 2 || parts[1] == "" {
+			allErrs = append(allErrs, field.Invalid(fldPath, method, "option cidr requires a parameter"))
+			return allErrs
+		}
+
+		err := validation.IsValidCIDR(nil, parts[1])
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath, method, fmt.Sprintf("invalid CIDR: %q", err)))
+		}
+	}
+
+	if option == "can-reach" || option == "interface" || option == "skip-interface" || option == "cidr" {
+		if len(parts) != 2 || parts[1] == "" {
+			allErrs = append(allErrs, field.Invalid(fldPath, method, fmt.Sprintf("option %s requires a parameter", option)))
+			return allErrs
+		}
+
+		if option == "interface" || option == "skip-interface" {
+			_, err := regexp.Compile(parts[1])
+			if err != nil {
+				allErrs = append(allErrs, field.Invalid(fldPath, method, fmt.Sprintf("invalid regex for option %s: %v", option, err)))
+			}
+		}
+	}
+
+	return allErrs
+}
+
+// IsValidMTU checks if the provided MTU is a valid positive integer.
+func IsValidMTU(mtu string) error {
+	if mtu == "" {
+		return fmt.Errorf("MTU cannot be empty")
+	}
+	mtuInt, err := strconv.Atoi(mtu)
+	if err != nil {
+		return fmt.Errorf("invalid MTU: %w", err)
+	}
+
+	if mtuInt <= 0 {
+		return fmt.Errorf("MTU must be a positive integer")
+	}
+
+	return nil
+}
+
+func ValidateNetworkConfigAutoscaling(autoscaling *apiscalico.AutoScaling, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if autoscaling == nil {
+		return allErrs
+	}
+	if !sets.New(apiscalico.AutoscalingModeClusterProportional, apiscalico.AutoscalingModeVPA, apiscalico.AutoscalingModeStatic).Has(autoscaling.Mode) {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("mode"), autoscaling.Mode, fmt.Sprintf("unsupported value %q for mode, supported values are [%q, %q, %q]", autoscaling.Mode, apiscalico.AutoscalingModeClusterProportional, apiscalico.AutoscalingModeVPA, apiscalico.AutoscalingModeStatic)))
+	}
+	if autoscaling.Resources != nil {
+		if autoscaling.Resources.Node != nil {
+			for name, quantity := range *autoscaling.Resources.Node {
+				switch name.String() {
+				case "cpu":
+					if quantity.IsZero() {
```

Comment:
Maybe we also want to check that the value is positive (`quantity.Sign()`). (Applies to the memory check as well.)
WDYT?